### PR TITLE
CMake refactoring for spack compatability

### PR DESCRIPTION
**Pull request by AdhocMan**
CMake changes to allow more fine grained control over library and python module installation location.
This is required for installation with spack.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -68,18 +77,34 @@ set(BIPP_GPU "OFF" CACHE STRING "GPU backend")
...
 if(BIPP_BUILD_TYPE)
-	set(CMAKE_BUILD_TYPE ${BIPP_BUILD_TYPE} CACHE STRING "Build type" FORCE)
+	set(CMAKE_BUILD_TYPE ${BIPP_BUILD_TYPE})
+endif()
+
+
+# Python install location
+if(BIPP_PYTHON)
+	set(Python_FIND_FRAMEWORK LAST) # Prefer Brew/Conda to Apple framework python
+	find_package(Python 3.6 COMPONENTS Interpreter Development REQUIRED)
```
Comment (orliac) [inline CMakeLists.txt:93]: Is it going to look specifically for Python 3.6?

Comment (AdhocMan) [reply CMakeLists.txt:93]: That's just the minimum version requirement. It's unchanged from where it was moved from in python/CMakeLists.txt.

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -42,6 +42,7 @@ add_executable(run_tests
   test_standard_synthesis_lofar.cpp
   test_domain_partition.cpp
 )
-target_link_libraries(run_tests PRIVATE ${BIPP_TEST_LIBRARIES} bipp_test)
+target_link_libraries(run_tests PRIVATE ${BIPP_TEST_LIBRARIES} bipp_objects ${BIPP_EXTERNAL_LIBS})
```
Comment (orliac) [inline tests/CMakeLists.txt:45]:
```suggestion
target_link_libraries(run_tests PRIVATE ${BIPP_TEST_LIBRARIES} bipp_objects ${BIPP_EXTERNAL_LIBS} ${BIPP_EXTERNAL_LIBS_PUBLIC})
```

File: python/CMakeLists.txt [hunk 3]
```diff
@@ -23,7 +20,7 @@ endif()
 
 
 pybind11_add_module(pybipp pybipp.cpp)
-target_link_libraries(pybipp PRIVATE bipp)
+target_link_libraries(pybipp PRIVATE bipp_objects)
```
Comment (orliac) [inline python/CMakeLists.txt:23]:
```suggestion
target_link_libraries(pybipp PRIVATE bipp_objects ${BIPP_EXTERNAL_LIBS_PUBLIC})
```

**Review by orliac — CHANGES_REQUESTED** [review]
I'm proposing a couple of changes without which BIPP does not compile on my side when asking MPI support.

**Review by orliac — APPROVED** [review]
Looks fine to me.

**Pull request merged**